src/NHibernate/AdoNet/ConnectionManager.cs

```diff
@@ -372,8 +378,7 @@ public DbCommand CreateCommand()
 		public void EnlistIfRequired(System.Transactions.Transaction transaction)
 		{
 			if (_connection == null ||
-				// Let user handles transaction if connection was supplied.
-				!_ownConnection ||
+				!ShouldAutoJoinTransaction ||
```

Comment:
If an user do not want auto-enlistment for a supplied connection, he will have to disable it with AutoJoinTransaction when building its session.